src/HoldRequestResultConsumerListener.php

```diff
@@ -328,6 +328,13 @@ protected function processListenerEvents()
                     'Exception thrown: ' . $exception->getMessage() .
                     ', Error code: ' . $exception->getCode()
                 );
+
+                if (strpos($exception->getMessage(), 'Operation timed out') !== false) {
```

Comment:
This logic is good! I wonder if maybe you should try to catch this exception in your other client-related code and then throw a timeout related exception (e.g. `ClientTimeoutException`). Also remember to throw the original exception if it's _not_ a client timeout.